packages/playground/src/utils/migration.ts

```diff
@@ -1,7 +1,52 @@
-import type { BaseModule } from "@threefold/grid_client/dist/es6/modules/base";
-import type { Contract } from "@threefold/tfchain_client";
+import type { Contract, ExtrinsicResult } from "@threefold/tfchain_client";
+
+import { BaseModule } from "../../../grid_client/dist/es6/modules/base";
+
+async function _migrateOldFullVMs(module: BaseModule) {
+  module.moduleName = "Fullvm"; // Load old deployments that deployed with `Fullvm` type.
+  const deploymentNames = await module._list();
+  const contracts = await Promise.all(
+    deploymentNames.map(deploymentName => module.getDeploymentContracts(deploymentName)),
+  );
+
+  const extrinsics: ExtrinsicResult<Contract>[] = [];
+
+  const loadUpdateExtrinsics = async () => {
+    return await Promise.all(
+      contracts.flat().map(async contract => {
+        const oldData = JSON.parse(contract.deploymentData || "{}") as unknown as {
+          type: string;
+          name: string;
+          projectName: string;
+        };
+
+        if (oldData.type === "vm") {
+          return;
+        }
+
+        oldData.type = "vm";
+
+        const extrinsic = await module.tfClient.contracts.updateNode({
+          id: +contract.contractID,
+          data: JSON.stringify(oldData),
+          hash: contract.deploymentHash,
+        });
+
+        extrinsics.push(extrinsic);
+        BaseModule.newContracts.push(contract);
+      }),
+    );
+  };
+
+  if (deploymentNames.length) {
+    await loadUpdateExtrinsics();
+    return await module.tfClient.applyAllExtrinsics<Contract>(extrinsics);
+  }
+}
 
 export async function migrateModule(module: BaseModule) {
+  await _migrateOldFullVMs(module);
+  module.moduleName = "vm";
```

Comment:
why do we set moduleName here to "vm"?

Reply:
we changed the `moduleName` to `Fullvm` in the `_migrateOldFullVMs` function, so we need to return it back to `vm` to list the other deployments

Reply:
we discussed this private ..